src/AdvancedBracketBalance/advancedBracketBalance.c

```diff
@@ -2,75 +2,52 @@
 #include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
-bool checkBrackets(int counter, char *stringWithBrackets) {
-  if (counter == 0) {
-    return false;
-  }
-  Stack *stack = newStack();
-  char tmp = ' ';
-  for (int i = 0; i < counter; i++) {
-    tmp = stringWithBrackets[i];
-    // проверка что стек не пустой
-    if ((tmp == '(') || (tmp == '{') || (tmp == '[')) {
-      push(stack, tmp);
+bool checkBrackets(char* stringWithBrackets)
+{
+    bool isBalanced = true;
+    Stack* stack = newStack();
+    char tmp = ' ';
+    int length = strlen(stringWithBrackets);
+    for (int i = 0; i < length; i++) {
+        tmp = stringWithBrackets[i];
```

Comment:
Безопаснее делать так:
```suggestion
        char ch = stringWithBrackets[i];
```
Чтобы переменная не была общей между итерациями цикла. 
Ну и название `tmp` прям совсем не информативно. `ch` не сказать, что шибко лучше, но хотя бы понятно, что это некий символ